x-pack/plugins/ml/public/application/jobs/new_job/pages/components/datafeed_step/components/reset_query/reset_query.tsx

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC, useContext, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import {
+  EuiButtonEmpty,
+  EuiConfirmModal,
+  EuiOverlayMask,
+  EuiCodeBlock,
+  EuiSpacer,
+} from '@elastic/eui';
+import { JobCreatorContext } from '../../../job_creator_context';
+
+const DEFAULT_QUERY = {
+  bool: {
+    must: [
+      {
+        match_all: {},
+      },
+    ],
+  },
+};
+const DEFAULT_QUERY_STRING = JSON.stringify(DEFAULT_QUERY, null, 2);
+
+export const ResetQueryButton: FC = () => {
+  const { jobCreator, jobCreatorUpdate } = useContext(JobCreatorContext);
+  const [confirmModelVisible, setConfirmModelVisible] = useState(false);
```

Comment:
Suppose this should be `[confirmModalVisible, setConfirmModalVisible]` with `a` instead of `e`.